src/ui/public/index_patterns/_index_pattern.js

```diff
@@ -293,8 +293,19 @@ define(function (require) {
       };
 
       self._fetchFields = function () {
+        var existingFieldsByName = self.fields.byName;
+
         return mapper.getFieldsForIndexPattern(self, true)
         .then(function (fields) {
+
+          // copy over kibana-added properties from existing fields
+          fields.forEach(function (field) {
```

Comment:
Because all of the other values are pulled from elasticsearch. In this context the `field` value is a fresh field mapping as read from the mappings, so `field.doc_values` reflects the current setting in the elasticsearch mapping. `field.exclude` on the otherhand is empty, because it is not set by the es mapping but by kibana.